### PR TITLE
PLT-848 Add tutorial intro screens and popover tips

### PR DESCRIPTION
**Pull request by jwilander**
All new and existing users will go through the tutorial when it goes in. Thanks to @asaadmahmoodspin for the UI work

### CODE REVIEW

File: web/react/components/post_list.jsx [hunk 1]
```diff
@@ -264,13 +264,15 @@ export default class PostList extends React.Component {
...
     scrollToBottom(force) {
         this.isUserScroll = false;
-        var postHolder = $(ReactDOM.findDOMNode(this.refs.postlist));
         if ($('#new_message_' + this.props.channelId)[0] && !this.userHasSeenNew && !force) {
             $('#new_message_' + this.props.channelId)[0].scrollIntoView();
         } else {
-            postHolder.addClass('hide-scroll');
-            postHolder[0].scrollTop = postHolder[0].scrollHeight;
-            postHolder.removeClass('hide-scroll');
+            const postHolder = $(ReactDOM.findDOMNode(this.refs.postlist));
```
Comment (crspeller) [inline web/react/components/post_list.jsx]: This is going to break when my center channel changes go in.

File: web/react/components/post_list_container.jsx [hunk 2]
```diff
@@ -2,7 +2,14 @@
 // See License.txt for license information.
 
 const PostList = require('./post_list.jsx');
+const TutorialIntroScreens = require('./tutorial/tutorial_intro_screens.jsx');
```
Comment (crspeller) [inline web/react/components/post_list_container.jsx]: Code in this file should probably be moved to ChannelView after my changes go in.

File: web/react/components/tutorial/tutorial_intro_screens.jsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    constructor(props) {
+        super(props);
+
+        this.handleNext = this.handleNext.bind(this);
+        this.createScreen = this.createScreen.bind(this);
+
+        this.state = {currentScreen: 0};
+    }
+    componentDidMount() {
+        const height = $(window).outerHeight() - 120;
```
Comment (crspeller) [inline web/react/components/tutorial/tutorial_intro_screens.jsx]: Why is this magic necessary? Can't it just auto fill the height?

File: web/react/components/post_list_container.jsx (hunk 2 above)
Comment (jwilander) [inline web/react/components/post_list_container.jsx]: Will do after they go in

File: web/react/components/post_list.jsx (hunk 1 above)
Comment (jwilander) [inline web/react/components/post_list.jsx]: Will fix when it goes in

File: web/react/components/tutorial/tutorial_intro_screens.jsx (hunk 3 above)
Comment (jwilander) [inline web/react/components/tutorial/tutorial_intro_screens.jsx]: Ah yeah, that's some of Asaad's code. I'll see if I can get rid of the jquery stuff

**Comment by jwilander** [comment]
Ready for re-review @crspeller

**Comment by jwilander** [comment]
Ok, now ready for re-review @crspeller

**Comment by crspeller** [comment]
+1 Looks good!

**Pull request merged**